src/bentoml/_internal/frameworks/diffusers.py

```diff
@@ -525,7 +498,10 @@ def import_model(
             )
 
     elif pipeline_class:
-        _try_import_huggingface_hub()
+        if not is_huggingface_hub_available():
+            raise MissingDependencyException(
+                "'huggingface_hub' is required in order to download pretrained models, install with 'pip install huggingface-hub'. For more information, refer to https://huggingface.co/docs/huggingface_hub/quick-start",
+            ) from None
```

Comment:
still not sure why we need to check this here. Since we don't use huggingface_hub here?

Reply:
`pipeline_class.download()` uses `snapshot_download` and `hf_hub_download` under the hood

Reply:
Then it should be their responsibility to check whether huggingface_hub is available or not right?

Reply:
hmm, I don't see that they check for `huggingface_hub` availability in their code. I believe adding this check on our end is not too costly performance-wise and also doesn't take away code legibility.

Reply:
Actually you don't have to check this at all, since `huggingface-hub` is a core dependency of diffusers.

See https://github.com/huggingface/diffusers/blob/dd9a5caf61f04d11c0fa9f3947b69ab0010c9a0f/setup.py#L237

Reply:
I see, I will remove the check. It seems for transformers, huggingface-hub is also a core dependency so I will remove the check in transformers as well.